src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala

```diff
@@ -322,6 +323,8 @@ class AstToCpgTests extends WordSpec with Matchers {
       whileStmt.check(1, _.value2(NodeKeys.CODE), expectations = "while (x < 1)")
       whileStmt.check(1, whileStmt => whileStmt.value2(NodeKeys.PARSER_TYPE_NAME), expectations = "WhileStatement")
 
+      isLinkedToCondition(whileStmt, "x < 1")
```

Comment:
Please do not introduce a different testing pattern.

Define a method `expandCondition` analog to `expandAst` and than use `checkForSingle`.

Also please add an extra test an extra test dedicated to checking the condition.